index.js

```diff
@@ -31,15 +31,23 @@ function defaultArrayMerge(target, source, optionsArgument) {
 }
 
 function mergeObject(target, source, optionsArgument) {
+    var clone = optionsArgument && optionsArgument.hasOwnProperty('clone') 
+        ? optionsArgument.clone : false
     var destination = {}
     if (isMergeableObject(target)) {
         Object.keys(target).forEach(function (key) {
-            destination[key] = target[key]
+            var val = target[key]
+            if (clone && typeof val === 'object') 
```

Comment:
I'm guessing that this should be `isMergeableObject(val)` - but it would be good to have some tests covering dates/arrays first to make sure.